### PR TITLE
common/ceph_timer: set thread name

### PR DESCRIPTION
**Pull request by ddiss**

### CODE REVIEW

**Comment by ddiss** [comment]
@adamemerson looks like you've done some changes in this area recently. Any chance you'd be able to take a look?

File: src/common/ceph_timer.h [hunk 1]
```diff
@@ -145,6 +146,7 @@ class timer {
 public:
   timer() : suspended(false) {
     thread = std::thread(&timer::timer_thread, this);
+    ceph_pthread_setname(thread.native_handle(), "ceph_timer");
```
Comment (adamemerson) [inline src/common/ceph_timer.h:149]: You might want to use `make_named_thread` from common/Thread.h (but you don't have to, either way is fine.)

File: src/test/common/CMakeLists.txt [hunk 2]
```diff
@@ -335,5 +335,4 @@ target_link_libraries(unittest_rabin_chunk global ceph-common)
 add_ceph_unittest(unittest_rabin_chunk)
 
 add_executable(unittest_ceph_timer test_ceph_timer.cc)
-target_link_libraries(unittest_rabin_chunk GTest::GTest)
```
Comment (adamemerson) [inline src/test/common/CMakeLists.txt:338]: Good catch.

File: src/common/ceph_timer.h (hunk 1 above)
Comment (ddiss) [reply src/common/ceph_timer.h:149]: Thanks a lot for the review :-)
I saw that there are a few other wrappers to handle naming, but figured this was less intrusive. Happy to change it if somebody feels strongly about it though.

**Comment by tchaikov** [comment]
@ddiss i change the title of this PR to "common/ceph_timer: set thread name" as i think the change of thread name is far more important than the cmake cleanup. so when people searching for the relevant change, he/she is more likely to find the right change.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2020-05-21_10:34:02-rados-wip-kefu-testing-2020-05-21-1652-distro-basic-smithi/

the failures are either tracked or fixed by

- https://tracker.ceph.com/issues/44595
- https://tracker.ceph.com/issues/45647
- https://tracker.ceph.com/issues/45631
- https://tracker.ceph.com/issues/44990
- #35139

**Pull request merged**